librz/include/rz_demangler.h

```diff
@@ -0,0 +1,48 @@
+// SPDX-FileCopyrightText: 2021 RizinOrg <info@rizin.re>
+// SPDX-FileCopyrightText: 2021 deroad <wargio@libero.it>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#ifndef RZ_DEMANGLER_H
+#define RZ_DEMANGLER_H
+#include <rz_types.h>
+#include <rz_list.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+RZ_LIB_VERSION_HEADER(rz_demangler);
+
+typedef struct rz_demangler_plugin_t {
+	const char *language; ///< demangler language
+	const char *author; ///< demangler author
+	const char *license; ///< demangler license
+	RZ_OWN char *(*demangle)(RZ_NONNULL const char *symbol); ///< demangler method to resolve the mangled symbol
+} RzDemanglerPlugin;
+
+typedef struct rz_demangler_t {
+	RzList *plugins;
+} RzDemangler;
+
+typedef bool (*RzDemanglerIter)(const RzDemanglerPlugin *plugin, void *data);
+
+#define rz_demangler_plugin_demangle(x, y) ((x) && RZ_STR_ISNOTEMPTY(y) ? (x)->demangle(y) : NULL)
+
+RZ_API RZ_OWN char *rz_demangler_java(RZ_NULLABLE const char *symbol);
+RZ_API RZ_OWN char *rz_demangler_cxx(RZ_NONNULL const char *symbol);
+RZ_API RZ_OWN char *rz_demangler_objc(RZ_NONNULL const char *symbol);
+RZ_API RZ_OWN char *rz_demangler_rust(RZ_NONNULL const char *symbol);
+RZ_API RZ_OWN char *rz_demangler_msvc(RZ_NONNULL const char *symbol);
+
+RZ_API RZ_OWN RzDemangler *rz_demangler_new();
```

Comment:
```suggestion
RZ_API RZ_OWN RzDemangler *rz_demangler_new(void);
```
?? i don't remember if it's the same or not..

Reply:
yes, it is the same. i can add that.